ports/servoshell/minibrowser.rs

```diff
@@ -153,7 +154,7 @@ impl Minibrowser {
 
         match browser.current_url_string() {
             Some(location) if location != self.location.get_mut() => {
-                self.location = RefCell::new(location.to_owned());
+                self.location = RefCell::new(self.location.get_mut().to_string());
```

Comment:
this change means the location never updates when you click a link :(

Reply:
Ahh, how did I miss that?! 😅 I'll take a step back and rethink how we can resolve this without causing any issues. I'm going to convert this PR to a draft for now. Thanks!

Reply:
no worries! btw don’t be afraid to change how the underlying events work or add a new event, if their behaviour is inadequate. and be aware of the edge case of navigating to http 204 :p

```js
// node test.js, then go to http://localhost:8000
const http = require("node:http");
const server = http.createServer((req, res) => {
    res.statusCode = 204;
    res.end();
});
server.listen(8000);
```

Reply:
@delan as we discussed #30696 
I think this PR is ready for review now.